packages/swapper/src/swappers/cow/cowBuildTrade/cowBuildTrade.ts

```diff
@@ -0,0 +1,148 @@
+import { CowSwapQuoteResponse } from '../types'
+import {
+  COW_SWAP_VAULT_RELAYER_ADDRESS,
+  DEFAULT_APP_DATA,
+  DEFAULT_SOURCE,
+  ORDER_KIND_SELL
+} from '../utils/constants'
+import { cowService } from '../utils/cowService'
+import { getNowPlusThirtyMinutesTimestamp, getUsdRate } from '../utils/helpers/helpers'
+
+export async function cowBuildTrade(
+  deps: CowSwapperDeps,
+  input: BuildTradeInput
+): Promise<Trade<'eip155:1'>> {
+  try {
+    const { sellAsset, buyAsset, sellAmount, sellAssetAccountNumber, wallet } = input
+    const { adapter, feeAsset } = deps
+
+    const { assetReference: sellAssetErc20Address, assetNamespace: sellAssetNamespace } =
+      fromAssetId(sellAsset.assetId)
+    const { assetReference: buyAssetErc20Address, assetNamespace: buyAssetNamespace } = fromAssetId(
+      buyAsset.assetId
+    )
+
+    if (buyAssetNamespace !== 'erc20' || sellAssetNamespace !== 'erc20') {
+      throw new SwapError('[cowBuildTrade] - Both assets need to be ERC-20 to use CowSwap', {
+        code: SwapErrorTypes.UNSUPPORTED_PAIR,
+        details: { buyAssetNamespace, sellAssetNamespace }
+      })
+    }
+
+    const receiveAddress = await adapter.getAddress({ wallet })
+    const normalizedSellAmount = normalizeAmount(sellAmount)
+
+    /**
+     * /v1/quote
+     * params: {
+     * sellToken: contract address of token to sell
+     * buyToken: contractAddress of token to buy
+     * receiver: receiver address can be defaulted to "0x0000000000000000000000000000000000000000"
+     * validTo: time duration during which quote is valid (eg : 1654851610 as timestamp)
+     * appData: appData for the CowSwap quote that can be used later, can be defaulted to "0x0000000000000000000000000000000000000000000000000000000000000000"
+     * partiallyFillable: false
+     * from: sender address can be defaulted to "0x0000000000000000000000000000000000000000"
+     * kind: "sell" or "buy"
+     * sellAmountBeforeFee / buyAmountAfterFee: amount in base unit
+     * }
+     */
+    const quoteResponse: AxiosResponse<CowSwapQuoteResponse> =
+      await cowService.post<CowSwapQuoteResponse>(`${deps.apiUrl}/v1/quote/`, {
+        sellToken: sellAssetErc20Address,
+        buyToken: buyAssetErc20Address,
+        receiver: receiveAddress,
+        validTo: getNowPlusThirtyMinutesTimestamp(),
+        appData: DEFAULT_APP_DATA,
+        partiallyFillable: false,
+        from: receiveAddress,
+        kind: ORDER_KIND_SELL,
+        sellAmountBeforeFee: normalizedSellAmount
+      })
+
+    const {
+      data: { quote }
+    } = quoteResponse
+
+    const buyCryptoAmount = bn(quote.buyAmount).div(bn(10).exponentiatedBy(buyAsset.precision))
+    const sellCryptoAmount = bn(quote.sellAmount).div(bn(10).exponentiatedBy(sellAsset.precision))
+    const rate = buyCryptoAmount.div(sellCryptoAmount).toString()
+
+    const [feeDataOptions, feeAssetUsdRate, sellAssetUsdRate] = await Promise.all([
+      adapter.getFeeData({
```

Comment:
same comment here regarding contract `inputData`. also not for this pr probably, but we should be using eip1559 fees if  supported for these trades as well.